packages/flutter/lib/src/painting/box_decoration.dart

```diff
@@ -206,6 +210,10 @@ class BoxDecoration extends Decoration {
   /// [BoxShape.rectangle].
   ///
   /// {@macro flutter.painting.BoxDecoration.clip}
+  @Deprecated(
+    'For BoxShape.circle, migrate to ShapeDecoration + CircleBorder(). '
```

Comment:
For better clarity let's say something `replace BoxDecoration(shape: BoxShape.circle) with ShapeDecoration(shape: const CircleBorder()` or something. Actually we might even be able to automigrate this, cc @Piinks.

(Before we do an automigration we really should see what the performance impact of this change is.)

Reply:
Is it possible to verify the performance impact **now**? Ignoring some tests?

Reply:
you can run some of the benchmarks locally before and after, sure. see the `dev/benchmarks` directory.